Stefan jum/add run yaml challenge (#324)
unikraft/docs: content/guides/challenges.mdx

You will get HP for submitting an issue or a solution that works.
Solutions are submitted as pull requests.

> Take some time to learn [how to work with `git`](https://gitimmersion.com/), and on [how to open pull requests](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github).

Reviewer: ```suggestion
Take some time to learn [how to work with Git](https://gitimmersion.com/), and [how to open pull requests (PRs) on GitHub](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github).
```

   30-50 HP are awarded for each submitted application, depending on its complexity.

> 1. **Add support for running multiple application in [`app-elfloader`](https://github.com/unikraft/app-elfloader/)**

Reviewer: ```suggestion
1. **Add support for running applications with [`app-elfloader`](https://github.com/unikraft/app-elfloader/)**
```

>     You need to update the [`run.yaml`](https://github.com/unikraft/app-elfloader/blob/stable/scripts/run.yaml) file inside [`app-elfloader`](https://github.com/unikraft/app-elfloader/) to add support for multiple applications from the [`dynamic-apps-repo`](://github.com/unikraft/dynamic-apps/).

Reviewer: ```suggestion
    Update the [`run.yaml`](https://github.com/unikraft/app-elfloader/blob/stable/scripts/run.yaml) file inside [`app-elfloader`](https://github.com/unikraft/app-elfloader/) to add support for existing applications in [the `dynamic-apps` repository](://github.com/unikraft/dynamic-apps/).
```

>     Follow the already existing examples for `helloworld` and `Nginx`.

Reviewer: ```suggestion
    Follow the already existing examples for `helloworld` and `nginx`.
```

>     You will need to add a new section to the `run.yaml` file, that will look like this:

Reviewer: ```suggestion
    Add a new section to the `run.yaml` file, that will look like this:
```

    ```yaml
    - name: app-name
      rootfs: path-to-dynamic-apps-app-directory
      command: run-command
      memory: amount of memoty needed
      networking: True / False depending on the application
    ```

>     To test your changes, from the `app-elfloader` directory, first run the `generate.py` script (`./scripts/generate.py`).

Reviewer: ```suggestion
    To test your changes, while insde the `app-elfloader` directory, first run the `generate.py` script (`./scripts/generate.py`).
```

>     If all went well, you should find some scripts in the `scripts/run/` directory named `qemu-x86_64-9pfs-<appname>.sh`.

Reviewer: ```suggestion
    If all went well, find scripts in the `scripts/run/` directory named `qemu-x86_64-9pfs-<appname>.sh`.
```

    Build the elfloader by running `./scripts/build/make-qemu-x86_64-9pfs.sh`, then run `./scripts/run/qemu-x86_64-9pfs-<appname>.sh` to see if the application is working.

>     If the run fails, it might be because of an issue in Unikraft, unrelated to your changes, so you can submit you pull request anyway, and also submit an issue with the error to get extra HP.

Reviewer: ```suggestion
    If the run fails, it might be because of an issue in Unikraft, unrelated to your changes.
    Submit you pull request anyway to [the `app-elfloader` repository](https://github.com/unikraft/app-elfloader).
    Also submit an issue with the error to [the `unikraft` core repository](https://github.com/unikraft/unikraft), and get extra HP.
```
